app/views/core/SubscribeModal.coffee

```diff
@@ -189,28 +180,27 @@ module.exports = class SubscribeModal extends ModalView
       return unless jqxhr # in case of cancellations
       @onSubscriptionError(jqxhr, failureMessage)
 
-  onClickStripeButton: ->
-    throw new Error("Can't use stripe on that product! Something went wrong.") unless @selectedProduct is @lifetimeProduct
+  onClickLifetimeButton: ->
```

Comment:
Might want to indicate via comment or otherwise why this part of lifetime flow is Stripe-only.  `onClickStripeLifetimeButton` maybe?